demos/python_demos/mri_segmentation.py

```diff
@@ -0,0 +1,303 @@
+            log.error("Please try to specify cpu extensions library path "
+                      "in sample's command line parameters using -l "
+                      "or --cpu_extension command line argument")
+            log.error("On CPU this is usually -l ${INTEL_CVSDK_DIR}/inference_engine/lib/centos_7.4/intel64/libcpu_extension_avx2.so")
+            log.error("You may need to build the OpenVINO samples directory for this library to be created on your system.")
+            log.error("e.g. bash ${INTEL_CVSDK_DIR}/inference_engine/samples/build_samples.sh will trigger the library to be built.")
+            log.error("Replace 'centos_7.4' with the pathname on your computer e.g. ('ubuntu_16.04')")
+            sys.exit(1)
+
+    assert len(net.inputs.keys()) == 1, "Sample supports only single input topologies"
+    assert len(net.outputs) == 1, "Sample supports only single output topologies"
+
+    """
+    Ask OpenVINO for input and output tensor names and sizes
+    """
+    input_blob = next(iter(net.inputs))  # Name of the input layer
+    out_blob = next(iter(net.outputs))   # Name of the output layer
+
+    batch_size, n_channels, height, width = net.inputs[input_blob].shape
+    batch_size, n_out_channels, height_out, width_out = net.outputs[out_blob].shape
+    net.batch_size = batch_size
+
+    # Load data
+    input_data, label_data, img_indicies = load_data()
+
+    # Loading model to the plugin
+    exec_net = plugin.load(network=net)
+    del net
+
+    if args.stats:
+        # Print the latency and throughput for inference
+        print_stats(exec_net, input_data, n_channels,
+                    batch_size, input_blob, out_blob, args)
+
+    """
+    OpenVINO inference code
+    input_blob is the name (string) of the input tensor in the graph
+    out_blob is the name (string) of the output tensor in the graph
+    Essentially, this looks exactly like a feed_dict for TensorFlow inference
+    """
+    # Go through the sample validation dataset to plot predictions
+    predictions = np.zeros((img_indicies.shape[0], n_out_channels,
+                            height_out, width_out))
+
+    for idx in range(0, img_indicies.shape[0], batch_size):
+
+        res = exec_net.infer(inputs={input_blob:
+                                     input_data[idx:(idx+batch_size),
+                                     :n_channels]})
+
+        # Save the predictions to array
+        predictions[idx:(idx+batch_size),] = res[out_blob]
+
+    if idx != (len(img_indicies)-1):  # Partial batch left in data
+        log.info("Partial batch left over in dataset.")
+
+    """
+    Evaluate model with Dice metric
+    """
+    for idx in range(img_indicies.shape[0]):
+        dice = dice_score(predictions[idx,0,:,:], label_data[idx,0,:,:])
+        log.info("Image #{}: Dice score = {:.4f}".format(img_indicies[idx], dice))
+
+    if args.plot:
+        plot_predictions(predictions, input_data, label_data, img_indicies, args)
```

Comment:
Please produce output (preferably nifti) so it can be viewed and evaluated externally.

Reply:
In line 52 I have the function "plot_predictions" which saves the images as pngs.  This is much easier than Nifti format since the casual user won't know how to the view a Nifti file and the advanced user will be able to save the NumPy to Nifti.